tests/tests_app/cli/test_connect.py

```diff
@@ -78,8 +78,11 @@ def fn(msg):
 
     assert _retrieve_connection_to_an_app() == (None, None)
 
+    connection_module._CLEAN_LIGHTNING_CONNECTION = True
+
 
 def test_connect_disconnect_cloud(monkeypatch):
+    connection_module._CLEAN_LIGHTNING_CONNECTION = False
```

Comment:
I suggest to use monkypatch to set this variable

```suggestion
    monkeypatch.setattr(connection_module, "_CLEAN_LIGHTNING_CONNECTION", False)
```

The benefit is that no matter how this test case exits (failure or not), monkeypatch will ensure that the variable gets reset to its original value. This helps avoid a potential cascade of failures in subsequent tests due to interactions through global variables.


Another suggestion is to monkeypatch the `_LIGHTNING_CONNECTION_FOLDER` not to be in the home folder, but instead to be under a tmpdir. For the same reason to 1) avoid interactions of tests through the file system 2) avoid leaving test artifacts behind on the developers system.

I believe these two suggestions combined would make the teardown logic

```py
    connection_module._CLEAN_LIGHTNING_CONNECTION = True
    connection_module._clean_lightning_connection()
```
at the end obsolete.

Reply:
Totally right @awaelchli

Reply:
Fixed, monkeypatching should be handled properly now. Thanks a lot @awaelchli